### PR TITLE
Feature/603 taker routing

### LINKED ISSUE #603: Catch faulty order strings and redirect to 404 page.
Compiling an order string should be done in the store. User should be redirected to OrderDetailWidget or a 404 order not found page.

Design will come up with 404 page.

### PR DESCRIPTION
**Pull request by piersss**
Fixes #603 

I also split the otc store into a make-otc and take-otc store.

@coltraneyan There's an `activeOrder` in the take-otc store you can use in the OrderDetailWidget. You can import it by using `useAppSelector` like so:

`const { activeOrder } = useAppSelector(selectTakeOtcReducer);`

Or you could pass it as a prop in `OrderDetail` to the `OrderDetailWidget`.

### CODE REVIEW

File: src/pages/OrderDetail/OrderDetail.tsx [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
 
 const OrderDetail: FC = () => {
+  const dispatch = useAppDispatch();
+  const { compressedOrder } = useParams<{ compressedOrder: string }>();
+
+  const { status } = useAppSelector(selectTakeOtcReducer);
+
+  useEffect(() => {
+    if (compressedOrder) {
+      dispatch(decompressAndSetActiveOrder(compressedOrder));
```
Comment (coltraneyan) [inline src/pages/OrderDetail/OrderDetail.tsx:19]:
```suggestion
      dispatch<any>(decompressAndSetActiveOrder(compressedOrder));
```

probably a better fix, but I get a type/overload error here and I used this to stop it

**Review by coltraneyan — CHANGES_REQUESTED** [review]

File: src/pages/OrderDetail/OrderDetail.tsx [hunk 2]
```diff
@@ -1,11 +1,37 @@
...
+    return (
+      <Page>
+        <InvalidOrder />
+      </Page>
+    );
+  }
+
   return (
     <Page>
       <OrderDetailWidget />
```
Comment (coltraneyan) [inline src/pages/OrderDetail/OrderDetail.tsx:37]: can now take in ```status``` and ```order``` props

File: src/pages/OrderDetail/OrderDetail.tsx (hunk 1 above)
Comment (piersss) [reply src/pages/OrderDetail/OrderDetail.tsx:19]: Strange, I don't get an error. Will add to fix things on your end.

File: src/pages/OrderDetail/OrderDetail.tsx (hunk 2 above)
Comment (piersss) [reply src/pages/OrderDetail/OrderDetail.tsx:37]: Cool!

**Review by coltraneyan — APPROVED** [review]

**Pull request merged**